src/txn/txn.c

```diff
@@ -138,6 +138,45 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
     }
 }
 
+/*
+ * __wt_txn_user_active --
+ *     Check whether there are any running user transactions.
+ */
+bool
+__wt_txn_user_active(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_SESSION_IMPL *session_in_list;
+    uint32_t i, session_cnt;
+    bool txn_active;
+
+    conn = S2C(session);
+    txn_active = false;
+
+    /*
+     * No lock is required because the session array is fixed size, but it may contain inactive
+     * entries. We must review any active session, so insert a read barrier after reading the active
+     * session count. That way, no matter what sessions come or go, we'll check the slots for all of
+     * the user sessions for active transactions when we started our check.
+     *
+     * Note that a new transactions may start on session we have already examined. Therefore, the
+     * caller needs to be aware of limitations of this function.
```

Comment:
Maybe we should clearly state this limitation in the method description?

Reply:
I moved the text to method description. See 50b8d55.